src/shogun/mathematics/linalg/LinalgBackendEigen.h

```diff
@@ -397,6 +478,36 @@ class LinalgBackendEigen : public LinalgBackendBase
 		result_eig = alpha * A_eig.col(i) + beta * b_eig;
 	}
 
+	/** Eigen3 vector add scalar method */
+	template <typename T>
+	void add_scalar_impl(SGVector<T>& a, T b) const
+	{
+		typename SGVector<T>::EigenVectorXtMap a_eig = a;
+		a_eig = a_eig.array() + b;
+	}
+
+	/** Eigen3 matrix add scalar method */
+	template <typename T>
+	void add_scalar_impl(SGMatrix<T>& a, T b) const
+	{
+		typename SGMatrix<T>::EigenMatrixXtMap a_eig = a;
```

Comment:
Note: A lot of methods can be merged by simply renaming `SGVector<T>::EigenVectorXtMap` to `SGVector<T>::EigenXtMap`and `SGMatrix<T>::EigenMatrixXtMap` to `SGMatrix<T>::EigenXtMap`, if it won't cause any confusion @karlnapf @vigsterkr